src/tools/tc_utils/tc_rmw/tc_rmw_wind_converter.cc

```diff
@@ -0,0 +1,242 @@
+  _foundVInInput = false;
+  _unitsU = "Unknown";
+  _uIndexMap.clear();
+  _vIndexMap.clear();
+
+  if (!_computeWinds) {
+    return;
+  }
+
+  VarInfo* data_info = (VarInfo*) 0;
+  for(int i_var = 0; i_var < _conf->get_n_data(); i_var++) {
+    data_info = _conf->data_info[i_var];
+    string varname = data_info->name_attr().string();
+    string varlevel = data_info->level_attr().string();
+    if (varname == _conf->u_wind_field_name.string()) {
+      _uIndexMap[varlevel] = i_var;
+    }
+    else if (varname == _conf->v_wind_field_name.string()) {
+      _vIndexMap[varlevel] = i_var;
+    }
+  }
+  // test for consistency
+  if (_uIndexMap.size() != _vIndexMap.size()) {
+    mlog << Error << " error uneven number of ugrid/vgrid, no wind conversion\n";
+    mlog << Error << _conf->u_wind_field_name.string() << " had " << _uIndexMap.size() << " inputs\n";
+    mlog << Error << _conf->v_wind_field_name.string() << " had " << _vIndexMap.size() << " inputs\n";
+    _computeWinds = false;
+  }
+  map<string,int>::const_iterator iu, iv;
+  for (iu=_uIndexMap.begin(), iv=_vIndexMap.begin(); iu!=_uIndexMap.end(); ++iu, ++iv) {
+    if (iu->first != iv->first) {
+      mlog << Error << "error ordering of ugrid/vgrid input levels not the same, not implemented, no wind conversions\n";
+      mlog << Error << "    "  << iu->first  << " " << iv->first << "\n";
+      _computeWinds = false;
+    }
+  }
+}
+
+////////////////////////////////////////////////////////////////////////
+
+void TCRMW_WindConverter::update_input(const std::string &variableName, const std::string &units) {
+  if (_computeWinds) {
+    if (variableName == _conf->u_wind_field_name.string()) {
+      _foundUInInput = true;
+      _unitsU = units;
+    }
+    else if (variableName == _conf->v_wind_field_name.string()) {
+      _foundVInInput = true;
+    }
+  }
+} 
+
+////////////////////////////////////////////////////////////////////////
+
+void TCRMW_WindConverter::append_nc_output_vars(map<string, vector<string> > &variable_levels,
+						map<string, string> &variable_long_names,
+						map<string, string> &variable_units) {
+  if (!_computeWinds) {
+    return;
+  }
+
+  if (_foundUInInput && _foundVInInput) {
+    variable_levels[_conf->tangential_velocity_field_name] = variable_levels[_conf->u_wind_field_name.string()];
+    variable_long_names[_conf->tangential_velocity_field_name] = _conf->tangential_velocity_long_field_name.string();
+    variable_units[_conf->tangential_velocity_field_name] = _unitsU;
+    variable_levels[_conf->radial_velocity_field_name] = variable_levels[_conf->u_wind_field_name.string()];
+    variable_long_names[_conf->radial_velocity_field_name] = _conf->radial_velocity_long_field_name.string();
+    variable_units[_conf->radial_velocity_field_name] = _unitsU;
+  }
+  else {
+    if (!_foundUInInput) {
+      mlog << Error << "\nTCWRMW_WindConverter::checkInputs() -> "
+	   << "field not found in input" << _conf->u_wind_field_name << "\n";
```

Comment:
Everywhere else in MET, all ERROR messages are followed by a call to exit(1) to return bad status.
Recommend either exiting here, or change the "Error" messages to "Warning" instead.

```suggestion
	   << "field not found in input \"" << _conf->u_wind_field_name << "\"\n";
```